spec/features/reservations_spec.rb

```diff
@@ -99,7 +99,45 @@
         expect(page).to have_content Time.zone.today.to_s(:long)
         expect(page).to have_content bad_due_date.to_s(:long)
         expect(page).to have_content @eq_model.name
-        expect(page).to have_content 'Create Reservation'
+        expect(page).to have_content 'Confirm Reservation'
+        expect(page).to have_content 'Please be aware of the following errors:'
+        find(:xpath, "//textarea[@id='reservation_notes']").set 'Because'
+        click_button 'Finalize Reservation'
+
+        # check that reservation was created with correct dates
+        query = Reservation.for_eq_model(@eq_model).for_reserver(reserver.id)
+        expect(query.count).to eq(1)
+        expect(query.first.start_date).to eq(Time.zone.today)
+        expect(query.first.due_date).to eq(bad_due_date)
+        expect(query.first.status).to eq('reserved')
+      end
+    end
+
+    shared_examples 'can edit valid reservation' do |reserver|
+      let(:reserver) { reserver }
+      str = reserver ? 'for other user successfully' : 'successfully'
+
+      it str do
+        visit root_path
```

Comment:
If I'm reading this right, isn't this test supposed to check that you can modify the reservation from the *confirm reservation* page (the `new_reservation_path`, I believe), not the catalog? I don't see any Capybara code that modifies the cart from the confirmation page, only from the catalog (which we tested before). Since we allow for JS responses we would have to reload that page after making our changes to run our expectations, but I'm not seeing that here.